src/main/java/com/the/hugging/team/utils/Dialogs.java

```diff
@@ -242,6 +246,271 @@ else if (dds.getText().length() != 11 &&
+        GridPane grid = new GridPane();
+        grid.setHgap(10);
+        grid.setVgap(10);
+        grid.setPadding(new Insets(20, 150, 10, 10));
+
+        TextField productName = new TextField();
+        productName.setPromptText("Name");
+        productName.setPrefWidth(250);
+
+        TextField nomenclature = new TextField();
+        nomenclature.setPromptText("Nomenclature");
+        nomenclature.setPrefWidth(250);
+
+        TextField quantityAmount = new TextField();
+        quantityAmount.setPromptText("Quantity amount");
+        quantityAmount.setPrefWidth(250);
+
+        TextField retailPrice = new TextField();
+        retailPrice.setPromptText("Retail price");
+        retailPrice.setPrefWidth(250);
+
+        TextField wholesalePrice = new TextField();
+        wholesalePrice.setPromptText("Wholesale price");
+        wholesalePrice.setPrefWidth(250);
+
+        TextField deliveryPrice = new TextField();
+        deliveryPrice.setPromptText("Delivery price");
+        deliveryPrice.setPrefWidth(250);
+
+        ChoiceBox<ProductCategory> category = new ChoiceBox<>(FXCollections.observableArrayList(productCategoryService.getAllProductCategories()));
+        StringConverter<ProductCategory> categoryConverter = new StringConverter<>() {
+            @Override
+            public String toString(ProductCategory productCategory) {
+                return productCategory.getName();
+            }
+
+            @Override
+            public ProductCategory fromString(String s) {
+                return null;
+            }
+        };
+        category.setConverter(categoryConverter);
+        category.getSelectionModel().selectFirst();
+
+        ChoiceBox<ProductQuantityType> quantityType = new ChoiceBox<>(FXCollections.observableArrayList(productQuantityTypeService.getAllProductQuantityTypes()));
+        StringConverter<ProductQuantityType> productQuantityTypeConverter = new StringConverter<>() {
+            @Override
+            public String toString(ProductQuantityType productQuantityType) {
+                return productQuantityType.getName();
+            }
+
+            @Override
+            public ProductQuantityType fromString(String s) {
+                return null;
+            }
+        };
+        quantityType.setConverter(productQuantityTypeConverter);
+        quantityType.getSelectionModel().selectFirst();
+
+        ChoiceBox<Shelf> shelf = new ChoiceBox<>(FXCollections.observableArrayList(storageService.getAllShelves()));
+        StringConverter<Shelf> shelfStringConverter = new StringConverter<>() {
+            @Override
+            public String toString(Shelf shelf) {
+                return shelf.getName();
+            }
+
+            @Override
+            public Shelf fromString(String s) {
+                return null;
+            }
+        };
+        shelf.setConverter(shelfStringConverter);
+        shelf.getSelectionModel().selectFirst();
+
+        if (product.getName() != null) {
```

Comment:
You can check if `porduct.getId() != null` instead of checking every property. They are all non-null values so they will be there.